pkgs/by-name/ar/archtika/package.nix

```diff
@@ -0,0 +1,57 @@
+{
+  lib,
+  stdenv,
+  buildNpmPackage,
+  importNpmLock,
+  symlinkJoin,
+  fetchFromGitHub,
+}:
+
+let
+  src = fetchFromGitHub {
+    owner = "archtika";
+    repo = "archtika";
+    rev = "v1.0.1";
+    hash = "sha256-+zZ2v2kYpJ12bJYURt4Ax5Mt3zgr+WQjnxLbAx0DKY0=";
+  };
+
+  web = buildNpmPackage {
+    name = "web-app";
+    src = "${src}/web-app";
+    npmDepsHash = "sha256-RTyo7K/Hr1hBGtcBKynrziUInl91JqZl84NkJg16ufA=";
+    npmFlags = [ "--legacy-peer-deps" ];
+    installPhase = ''
+      mkdir -p $out/web-app
+      cp package.json $out/web-app
+      cp -r node_modules $out/web-app
+      cp -r build/* $out/web-app
+      cp -r template-styles $out/web-app
+    '';
+  };
+
+  api = stdenv.mkDerivation {
+    name = "api";
+    src = "${src}/rest-api";
+    installPhase = ''
+      mkdir -p $out/rest-api/db/migrations
+      cp -r db/migrations/* $out/rest-api/db/migrations
+    '';
+  };
```

Comment:
It's best to split these in separate packages. (Even if their sources are the same monorepo)

Reply:
Could you elaborate on this? Are you saying that instead of just including the full package derivation in the output, you would also include the api and web packages as individual units?

Reply:
Yes. The full `symlinkJoin`ed package could probably also be put in the module, with individual package options for the web part and the API part that get combined in a single derivation with the module.